src/Components/UI/Modal/Signup.jsx

```diff
@@ -1,13 +1,63 @@
 import Button from "../Button/Button";
 import "./Signup.css";
 import InputTypeOne from "../Input/InputTypeOne";
-import { useModal } from "../../../Context";
+import { useAuth, useModal } from "../../../Context";
+import { useState } from "react";
+import axios from "axios";
 
 const Signup = () => {
-  const { setShowLoginModal, setShowSignupModal } = useModal();
+  const { setShowLoginModal, setShowSignupModal, setShowDiscardModal } =
+    useModal();
+  const [user, setUser] = useState({
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+  });
 
   const onSignupFormSubmitHandler = (e) => {
     e.preventDefault();
+    userSignup();
+    setShowSignupModal(false);
+    setUser({
```

Comment:
You can declare initial data and reuse it.

Reply:
Fixed